qualtran/_infra/bloq.py

```diff
@@ -166,7 +168,7 @@ def as_composite_bloq(self) -> 'CompositeBloq':
         bb, initial_soqs = BloqBuilder.from_signature(self.signature, add_registers_allowed=False)
         return bb.finalize(**bb.add_d(self, **initial_soqs))
 
-    def adjoint(self) -> 'Bloq':
+    def adjoint(self) -> 'GateWithRegisters':
```

Comment:
??

Reply:
This is because Bloq has no `__pow__ ` but we often do bloq.adjoint() ** 2 or similar.
Alternatives welcome.  I can also cast the bloq.adjoint() to GateWithRegisters?

Reply:
Actually, I fixed it in GateWithRegisters instead.  Reverted this.